### PR TITLE
Rename hir::ExprKind::Use to ::DropTemps and improve docs.

### PR DESCRIPTION
**Pull request by Centril**
Addresses https://github.com/rust-lang/rust/pull/60225#discussion_r279797691.

r? @oli-obk 

cc @eddyb @Manishearth

### CODE REVIEW

File: src/librustc/hir/mod.rs [hunk 1]
```diff
@@ -1488,10 +1488,13 @@ pub enum ExprKind {
     Cast(P<Expr>, P<Ty>),
     /// A type reference (e.g., `Foo`).
     Type(P<Expr>, P<Ty>),
-    /// Semantically equivalent to `{ let _t = expr; _t }`.
-    /// Maps directly to `hair::ExprKind::Use`.
-    /// Only exists to tweak the drop order in HIR.
-    Use(P<Expr>),
+    /// Wraps the expression in a terminating scope.
+    /// This makes it semantically equivalent to `{ let _t = expr; _t }`.
+    /// Beyond that, the semantics maps to `hair::ExprKind::Use`.
```
Comment (eddyb) [inline src/librustc/hir/mod.rs]: Hmm, I would remove this line, because it doesn't add much (AFAIK, lots of things map to `ExprKind::Use`).

Comment (Centril) [reply src/librustc/hir/mod.rs]: Done.

**Pull request merged**